commit 34646edc7884
Author: Andreas Pehrson <pehrsons@mozilla.com>
Date:   Mon Feb 26 10:32:29 2018 +0100

    Bug 1436117 - Keep aImage around until the WrappedI420Buffer is released. r=dminor
    
    MozReview-Commit-ID: GuBHFnPA3YQ
    
    --HG--
    extra : rebase_source : e5cc3c1d5e0f08279e8c40ebf03ef7e535cfe5e5
---
 media/webrtc/signaling/src/mediapipeline/MediaPipeline.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git media/webrtc/signaling/src/mediapipeline/MediaPipeline.cpp media/webrtc/signaling/src/mediapipeline/MediaPipeline.cpp
index 2bb7203f85bd0..cfaadaaa7ff73 100644
--- media/webrtc/signaling/src/mediapipeline/MediaPipeline.cpp
+++ media/webrtc/signaling/src/mediapipeline/MediaPipeline.cpp
@@ -57,6 +57,7 @@
 #include "webrtc/common_video/libyuv/include/webrtc_libyuv.h"
 #include "webrtc/common_video/include/video_frame_buffer.h"
 #include "webrtc/base/bind.h"
+#include "webrtc/base/keep_ref_until_done.h"
 
 #include "nsThreadUtils.h"
 
@@ -336,14 +337,13 @@ protected:
         uint32_t width = yuv->GetSize().width;
         uint32_t height = yuv->GetSize().height;
 
-        rtc::Callback0<void> callback_unused;
         rtc::scoped_refptr<webrtc::WrappedI420Buffer> video_frame_buffer(
           new rtc::RefCountedObject<webrtc::WrappedI420Buffer>(
             width, height,
             y, yStride,
             cb, cbCrStride,
             cr, cbCrStride,
-            callback_unused));
+            rtc::KeepRefUntilDone(aImage)));
 
         webrtc::VideoFrame i420_frame(video_frame_buffer,
                                       0, 0, // not setting timestamps
